### PR TITLE
Fix opensbi build

### PR DESCRIPTION
**Pull request by jlaitine**
This reverts commits which break opensbi building for PX4; our build environment doesn't run patches on top of it; also, we want to use upstream opensbi and not execute any patches on it.

Also, update the opensbi to the newer upstream version, which allows compiling with more strict compiler warning settings.

### CODE REVIEW

**Comment by pkarashchenko** [comment]
Ok. I will work with opensbi team to apply changes and restore compatibility.

File: arch/risc-v/src/Makefile [hunk 1]
```diff
@@ -38,10 +38,9 @@ INCLUDES += ${shell $(INCDIR) "$(CC)" $(ARCH_SRCDIR)$(DELIM)chip}
 INCLUDES += ${shell $(INCDIR) "$(CC)" $(ARCH_SRCDIR)$(DELIM)common}
 INCLUDES += ${shell $(INCDIR) "$(CC)" $(ARCH_SRCDIR)$(DELIM)$(ARCH_SUBDIR)}
 INCLUDES += ${shell $(INCDIR) "$(CC)" $(TOPDIR)$(DELIM)sched}
-ifeq ($(CONFIG_OPENSBI),y)
-INCLUDES += $(shell $(INCDIR) "$(CC)" $(ARCH_SRCDIR)$(DELIM)opensbi)
+#ifeq ($(CONFIG_OPENSBI),y)
```
Comment (pkarashchenko) [inline arch/risc-v/src/Makefile:41]:
```suggestion
ifeq ($(CONFIG_OPENSBI),y)
```

File: arch/risc-v/src/Makefile [hunk 2]
```diff
@@ -38,10 +38,9 @@ INCLUDES += ${shell $(INCDIR) "$(CC)" $(ARCH_SRCDIR)$(DELIM)chip}
...
 INCLUDES += $(shell $(INCDIR) "$(CC)" $(ARCH_SRCDIR)$(DELIM)opensbi$(DELIM)opensbi-3rdparty$(DELIM)include)
-endif
+#endif
```
Comment (pkarashchenko) [inline arch/risc-v/src/Makefile:43]:
```suggestion
endif
```

File: arch/risc-v/src/opensbi/Kconfig [hunk 3]
```diff
@@ -9,4 +9,4 @@ config OPENSBI
 	default n
 	---help---
 		Enable or disable Open Source Supervisor Binary Interface (OpenSBI) features
-		for RISC-V.
+                for RISC-V.
```
Comment (pkarashchenko) [inline arch/risc-v/src/opensbi/Kconfig:12]: Lets keep this change

File: arch/risc-v/src/mpfs/mpfs_opensbi.c [hunk 4]
```diff
@@ -23,27 +23,36 @@
...
 
 #include <nuttx/config.h>
-
 #include <assert.h>
 #include <errno.h>
 #include <stdint.h>
-
-#include "riscv_internal.h"
-#include "riscv_arch.h"
+#include <riscv_arch.h>
```
Comment (pkarashchenko) [inline arch/risc-v/src/mpfs/mpfs_opensbi.c:29]:
```suggestion

#include "riscv_arch.h"
```

File: arch/risc-v/src/mpfs/mpfs_opensbi.c [hunk 5]
```diff
@@ -23,27 +23,36 @@
...
+ * 'warning: "NULL" redefined'
+ */
+
+#ifdef NULL
+  #undef NULL
+#endif
+
 #include <sbi/sbi_types.h>
 #include <sbi/riscv_atomic.h>
+#include <sbi/riscv_asm.h>
```
Comment (pkarashchenko) [inline arch/risc-v/src/mpfs/mpfs_opensbi.c:46]: please check, but I think it is not needed here

File: arch/risc-v/src/mpfs/mpfs_opensbi.c [hunk 6]
```diff
@@ -23,27 +23,36 @@
...
 #include <sbi/riscv_io.h>
+#include <sbi/riscv_encoding.h>
```
Comment (pkarashchenko) [inline arch/risc-v/src/mpfs/mpfs_opensbi.c:48]: please check, but I think it is not needed here

File: arch/risc-v/src/mpfs/mpfs_opensbi.c [hunk 7]
```diff
@@ -23,27 +23,36 @@
...
 #include <sbi/sbi_hart.h>
 #include <sbi/sbi_console.h>
 #include <sbi/sbi_platform.h>
+#include <sbi/sbi_domain.h>
```
Comment (pkarashchenko) [inline arch/risc-v/src/mpfs/mpfs_opensbi.c:52]: please check, but I think it is not needed here

File: arch/risc-v/src/mpfs/mpfs_opensbi.c [hunk 8]
```diff
@@ -23,27 +23,36 @@
...
 #include <sbi/sbi_timer.h>
 #include <sbi/sbi_init.h>
+#include <sbi/sbi_scratch.h>
```
Comment (pkarashchenko) [inline arch/risc-v/src/mpfs/mpfs_opensbi.c:55]: please check, but I think it is not needed here

**Comment by eenurkka** [comment]
I guess what happens is that if one modifies mpfs_opensbi.c -file, and builds, the OpenSBI patch has no #defined values yet, so the conflict is opposite: those #ifndefs would need to be in both headers...

**Comment by pkarashchenko** [comment]
> 

Yes. I'm filling the issue in opensbi github and will start a discussion. Probably the best way will be to add `SBI_` or `OPEN_SBI_` prefix, so the build will be clean.
What I'm pointing here in comments there seems like there are many headers included that really are not needed for `mpfs_opensbi.c` to build

**Comment by pkarashchenko** [comment]
I filled https://github.com/riscv-software-src/opensbi/issues/243

File: arch/risc-v/src/mpfs/mpfs_opensbi.c (hunk 8 above)
Comment (jlaitine) [reply arch/risc-v/src/mpfs/mpfs_opensbi.c:55]: struct sbi_scratch comes from there

File: arch/risc-v/src/mpfs/mpfs_opensbi.c (hunk 7 above)
Comment (jlaitine) [reply arch/risc-v/src/mpfs/mpfs_opensbi.c:52]: that's right, not needed

File: arch/risc-v/src/mpfs/mpfs_opensbi.c (hunk 6 above)
Comment (jlaitine) [reply arch/risc-v/src/mpfs/mpfs_opensbi.c:48]: sbi mode macros such as PRV_S come from there

File: arch/risc-v/src/mpfs/mpfs_opensbi.c (hunk 5 above)
Comment (jlaitine) [reply arch/risc-v/src/mpfs/mpfs_opensbi.c:46]: true

File: arch/risc-v/src/opensbi/Kconfig (hunk 3 above)
Comment (jlaitine) [reply arch/risc-v/src/opensbi/Kconfig:12]: ok

File: arch/risc-v/src/Makefile (hunk 2 above)
Comment (jlaitine) [reply arch/risc-v/src/Makefile:43]: ok

**Comment by jlaitine** [comment]
I will remove the unnecessary includes, and push a cleanup patch on top.

Let's continue the cleanup after reaction from opensbi people

File: arch/risc-v/src/mpfs/mpfs_opensbi.c (hunk 4 above)
Comment (jlaitine) [reply arch/risc-v/src/mpfs/mpfs_opensbi.c]: sure

**Review by pkarashchenko — APPROVED** [review]
LGTM!

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**